modules/core/src/main/scala/org/tesselation/infrastructure/gossip/RumorHandler.scala

```diff
@@ -17,30 +19,44 @@ import org.tesselation.schema.peer.PeerId
 object RumorHandler {
 
   def fromReceivedRumorFn[F[_]: Async: KryoSerializer, A <: AnyRef: TypeTag: ClassTag](
-    f: ReceivedRumor[A] => F[Unit]
-  ): RumorHandler[F] = fromBiFn((id: PeerId, a: A) => f(ReceivedRumor(id, a)))
-
-  def fromFn[F[_]: Async: KryoSerializer, A <: AnyRef: TypeTag: ClassTag](
-    f: A => F[Unit]
-  ): RumorHandler[F] = fromBiFn((_: PeerId, a: A) => f(a))
-
-  def fromBiFn[F[_]: Async: KryoSerializer, A <: AnyRef: TypeTag: ClassTag](
-    f: (PeerId, A) => F[Unit]
-  ): RumorHandler[F] = {
+    rumorStorage: RumorStorage[F],
```

Comment:
I dont' think that `RumorStorage` should be an argument of any handler creating function. I think that we should use kleisli functionality and pass the storage along with the rumor, so that in places where you're creating handlers you don't need to depend on the `RumorStorage`.

Basically we would replace this:
`type RumorHandler[F[_]] = Kleisli[OptionT[F, *], Rumor, Unit]`
with this:
`type RumorHandler[F[_]] = Kleisli[OptionT[F, *], (Rumor, RumorStorage), Unit]`